cs-minibot-platform-src/src/main/java/simulator/simbot/Dijkstras.java

```diff
@@ -0,0 +1,199 @@
+        int width = maze[0].length;
+        for(int i = 1; i < height-1; i++) {
+            for(int j = 1; j < width -1; j++) {
+                if(maze[i][j].value == 0) {
+                    maze[i][j].neighbors = generateNeighbors(i, j, maze);
+                }
+            }
+        }
+        return maze;
+    }
+
+    public static Node getNodeWithLowestCost(HashSet<Node> unsettledNodes) {
+        int lowestCost = Integer.MAX_VALUE;
+        Node res = null;
+        for(Node n: unsettledNodes) {
+            if(n.cost < lowestCost) {
+                lowestCost = n.cost;
+                res = n;
+
+            }
+        }
+        return res;
+    }
+
+    public static void processNeighbors(Node currentNode) {
+        for(Node n: currentNode.neighbors) {
+            if(!settledNodes.contains(n)) {
+                int newCost = currentNode.cost + 1;
+                if(newCost < n.cost) {
+
+                    n.cost = newCost;
+                    predecessors.put(n, currentNode);
+                }
+                unsettledNodes.add(n);
+            }
+        }
+    }
+    public static LinkedList<Node> getPath(Node target) {
+        LinkedList<Node> path = new LinkedList<Node>();
+        Node step = target;
+        // check if a path exists
+        if (predecessors.get(step) == null) {
+            return null;
+        }
+        path.add(step);
+        while (predecessors.get(step) != null) {
+            step = predecessors.get(step);
+            path.add(step);
+
+
+        }
+        // Put it into the correct order
+        Collections.reverse(path);
+        return path;
+    }
+
+    public static int execute(Node[][] maze) {
+        Node sourceNode = maze[1][1];
+        unsettledNodes.add(sourceNode);
+        Node curr = sourceNode;
+        //predecessors.put(curr, null);
+        Node currentNode = null;
+        while(!unsettledNodes.isEmpty()) {
+            currentNode = getNodeWithLowestCost(unsettledNodes);
+
+            if(currentNode.isEnd){
+                predecessors.put(currentNode, curr);
+
+                break;
+            }
+            //path[currentNode.getX()][currentNode.getY()] = 1;
+            unsettledNodes.remove(currentNode);
+            settledNodes.add(currentNode);
+            //System.out.println("PUT");
+            curr = currentNode;
+            processNeighbors(currentNode);
+        }
+        //System.out.println("LAST CURRENTNODE " + currentNode.getX() + " " + currentNode.getY());
+
+        int width = maze.length;
+        int height = maze[0].length;
+        return maze[width-2][height-2].cost;
+    }
+
+    public static int answer(int[][] maze) {
+        int min = Integer.MAX_VALUE;
+
+        int[][] x = generateWrappedMaze(maze);
+        Node[][] y = initializeNodeMatrix(x);
+        Node[][] z = populateNeighbors(y);
+        int res = execute(z);
+        min = res < min ? res : min;
+
+
+        return min;
+    }
+    public static class Node {
```

Comment:
why is this class static? also I'm not sure it needs to be public (maybe it does, but hard to tell in code review)